flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/ExpressionParser.scala

```diff
@@ -186,7 +186,7 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
       }
 
   lazy val singleQuoteStringLiteral: Parser[Expression] =
-    ("'" + """([^'\p{Cntrl}\\]|\\[\\'"bfnrt]|\\u[a-fA-F0-9]{4})*""" + "'").r ^^ {
+    ("'" + """([^'\p{Cntrl}]|\\[\\'"bfnrt]|\\u[a-fA-F0-9]{4})*""" + "'").r ^^ {
```

Comment:
I'm not 100% sure if this is the correct fix. The regular expression has been adapted from Scala's `JavaTokenParsers`. Maybe we just need to update it as done [here](https://github.com/scala/scala-parser-combinators/blob/1.1.x/shared/src/main/scala/scala/util/parsing/combinator/JavaTokenParsers.scala)?

Reply:
I don't know if modifying the scala library will have a bigger impact. In fact, looking at this line, we are already customizing (expanding) this regular expression. My personal suggestion is that we only modify this regular expression for single quotes. (At the moment we only found problems in the single quotes scenario), and other forms have not been found, and it has passed all test cases, so it can be said that this change has no effect on other current parsing cases. But if we don't modify it, then all parsing of `\d`, '\w' and other regulars beginning with a backslash are wrong.